### PR TITLE
Add HTTP GET and POST Request Function

### PR DESCRIPTION
**Pull request by YaikaRace**
I added the HTTP get and post functions to make it easier for developers to make HTTP requests, this allows developers to use just one line of code to make a request with the get and post method and receive the response as a dictionary

### CODE REVIEW

**Review by sirjoehighton — CHANGES_REQUESTED** [review]
Add Error Handling besides that looks good to me and ready to Squash and Merge

**Comment by YaikaRace** [comment]
I already made the changes, I hope you can review it

File: addons/godottoolkit/toolkit.gd [hunk 1]
```diff
@@ -73,3 +90,72 @@ func error_lookup(ERROR_CODE : int) -> Dictionary:
...
+	if RESULT_CODE in http_result_code_dictionary:
+		return http_result_code_dictionary[RESULT_CODE]
+	else:
+		return {"name": "RESULT_UNKNOWN", "description": "No description available."}
+
+## For making awaitable GET requests in one line of code
+func http_get_request(url: String, custom_headers: PackedStringArray = PackedStringArray()) -> Dictionary:
+	if not url:
+		push_error("[GodotToolkit] HTTPRequest URL Error: URL can't be empty")
+		printerr("[GodotToolkit] HTTPRequest URL Error: URL can't be empty")
```
Comment (sirjoehighton) [inline addons/godottoolkit/toolkit.gd:105]: Duplicate Error Print

File: addons/godottoolkit/toolkit.gd [hunk 2]
```diff
@@ -73,3 +90,72 @@ func error_lookup(ERROR_CODE : int) -> Dictionary:
...
+		var result_msg = "[GodotToolkit] HTTPRequest %s: %s" % [result_lookup.name, result_lookup.description]
+		push_error(result_msg)
+		printerr(result_msg)
+	return parsed_dict
+
+## For making awaitable POST requests in one line of code
+func http_post_request(url: String, custom_headers: PackedStringArray = PackedStringArray(), payload: String = "") -> Dictionary:
+	if not url:
+		push_error("[GodotToolkit] HTTPRequest URL Error: URL can't be empty")
+		printerr("[GodotToolkit] HTTPRequest URL Error: URL can't be empty")
```
Comment (sirjoehighton) [inline addons/godottoolkit/toolkit.gd:136]: Remove [GodotToolkit]

File: addons/godottoolkit/toolkit.gd [hunk 3]
```diff
@@ -73,3 +90,72 @@ func error_lookup(ERROR_CODE : int) -> Dictionary:
...
+	}
+	http.queue_free()
+	if err != OK:
+		var error = error_lookup(err)
+		var error_msg = "[GodotToolkit] HTTPRequest %s: %s" % [error.name, error.description]
+		push_error(error_msg)
+		printerr(error_msg)
+	if parsed_dict.result != OK:
+		var result_lookup = http_result_code_lookup(parsed_dict.result)
+		var result_msg = "[GodotToolkit] HTTPRequest %s: %s" % [result_lookup.name, result_lookup.description]
```
Comment (sirjoehighton) [inline addons/godottoolkit/toolkit.gd:127]: Remove [GodotToolkit]

File: addons/godottoolkit/toolkit.gd [hunk 4]
```diff
@@ -73,3 +90,72 @@ func error_lookup(ERROR_CODE : int) -> Dictionary:
...
+		"result": response[0],
+		"response_code": response[1],
+		"headers": response[2],
+		"body": response[3],
+		"json_body": JSON.parse_string(response[3].get_string_from_utf8())
+	}
+	http.queue_free()
+	if err != OK:
+		var error = error_lookup(err)
+		var error_msg = "[GodotToolkit] HTTPRequest %s: %s" % [error.name, error.description]
```
Comment (sirjoehighton) [inline addons/godottoolkit/toolkit.gd:122]: Remove [GodotToolkit]

File: addons/godottoolkit/toolkit.gd [hunk 5]
```diff
@@ -73,3 +90,72 @@ func error_lookup(ERROR_CODE : int) -> Dictionary:
...
```
Comment (sirjoehighton) [inline addons/godottoolkit/toolkit.gd:153]: Remove [GodotToolkit]

File: addons/godottoolkit/toolkit.gd [hunk 6]
```diff
@@ -73,3 +90,72 @@ func error_lookup(ERROR_CODE : int) -> Dictionary:
...
+		push_error(error_msg)
+		printerr(error_msg)
+	if parsed_dict.result != OK:
+		var result_lookup = http_result_code_lookup(parsed_dict.result)
+		var result_msg = "[GodotToolkit] HTTPRequest %s: %s" % [result_lookup.name, result_lookup.description]
```
Comment (sirjoehighton) [inline addons/godottoolkit/toolkit.gd:158]: Remove [GodotToolkit]

**Review by sirjoehighton — CHANGES_REQUESTED** [review]

**Comment by YaikaRace** [comment]
Done, I apologize, it's my first pull request

**Comment by sirjoehighton** [comment]
Awesome, thanks for making the nesscarry changes. Before i merge it have you tested it to make sure it works as intended?

**Comment by YaikaRace** [comment]
Yes I did tests, and in the tests everything worked correctly

**Pull request merged**